tests/test_advanced_filters.py

```
@@ -0,0 +1,2458 @@
+                "OR": [
+                    {"nonexistent_field": "value"},
+                    {"metadata": {"role": "test"}},
+                ]
+            }
+        },
+    )
+    assert response.status_code == 422
+
+
+@pytest.mark.asyncio
+async def test_not_logic_correctness(
```

Ruff (0.11.9)

Missing return type annotation for public function `test_not_logic_correctness`

Add return type annotation: `None`

(ANN201)